joomla/joomla-cms
# [4.0] Fix mail templates list display

Pull Request for Issue # .

### Summary of Changes

This Pull Request (PR) fixes the list display of mail templates so that nu duplicate items are shown anymore after having changed some settings (e.g. use custom subject) for a particular language.

It also makes filtering by language and saving the changed settings work again after it has stopped working since some previous PR.

### Testing Instructions

1. Apply the patch pf this PR on a clean 4.0-dev branch and make a new installation.
2. In backend, go to mail templates navigating as shown in following screenshot with the red marks:

![Snap 1](https://user-images.githubusercontent.com/7413183/65991903-6dcab180-e48e-11e9-8c51-4d69304b5e3c.png)

Result: The mail templates list shows 1 template for extension `com_config.test_mail`, which shows the language of the installation as only language.

![mail-templates-1](https://user-images.githubusercontent.com/7413183/66114830-ebe2a180-e5cf-11e9-8e1b-ff9e2a11e6b4.png)

3. Install some languages, publish their content languages and go to the mail templates list again.

Result: The mail template shows now flags for all published content languages.

![mail-templates-2](https://user-images.githubusercontent.com/7413183/66115182-98248800-e5d0-11e9-8d2a-b17146e2128b.png)

4. Open the filter options and check that there are filters by language and extensions.

Result:

![mail-templates-3](https://user-images.githubusercontent.com/7413183/66115291-d15cf800-e5d0-11e9-8e5f-24e4bc5ff4ce.png)

5. Filter by some language.

Result: No item found. Reason: There has not been made any adjustment to a mail template for a language.

![mail-templates-4](https://user-images.githubusercontent.com/7413183/66161534-cd5ec380-e62c-11e9-9132-82012c28b86a.png)

6. Reset the filter so that the item is shown again.

7. Click on a language flag to edit the mail template for that particular language and make some change in the edit view, e.g. enable custom subject.

![mail-templates-5](https://user-images.githubusercontent.com/7413183/66115457-2ac52700-e5d1-11e9-81a4-d6de46815f69.png)

8. Save and close.

9. Filter again by language and select the language for which you have edited the mail template parameters just before.

Result: The item is shown.

![mail-templates-6](https://user-images.githubusercontent.com/7413183/66115542-547e4e00-e5d1-11e9-855f-6ae3e72a0d00.png)

### Expected result

See testing instructions above.

### Actual result

After having edited and saved the mail template for one language, the list shows **two (in case if no language installed) or maybe more (in case if some languages installed) records** shown with ID "com_config.test_mail".

![Unbenannt-4](https://user-images.githubusercontent.com/7413183/66003298-fb65cb80-e4a5-11e9-9648-b84306896357.png)

Changing mail subject and content for a particular language doesn't work, i.e. changes are not there when editing again.

### Additional information

When using the language filter for "All languages", there will never be an item found because there is no item with language='*' in database, and such will never be created. This could be fixed by either removing this option from the language filter (but this would not be consistent with all other language filters anywhere else) or by showing the master item as usual, i.e. without language filter applied.

But this should be done with another PR if necessary/desired.

### Documentation Changes Required

None.
